### ISSUE TITLE
Segmentation fault with invalid Elm c

### ISSUE THREAD

**Issue by thistent** [helix-editor/helix#6981]
### Summary

I wrote this small invalid Elm code using **helix 23.03** inside a tmux session:

```elm
module Test exposing (..)

hi =
  case Expr of
```

Pressing enter (and possibly tab) after `of` caused a segmentation fault! If the language server was running, I found that I would sometimes have to delete all but one space between the `=` and the `case` and then try to press enter at the end of the line to get the same error.

After **helix** quit, escape codes were being printed when hovering over the effected pane in tmux.

![helix-tmux-error](https://user-images.githubusercontent.com/614897/236636043-1e891865-8188-4bda-8108-6cd8a252c8ad.png)

The bug didn't seem to occur in when running **hx** on kitty without tmux.

### Reproduction Steps

 

I tried this:

1. `touch Test.elm`
2. `hx Test.elm` 
  Note: the language server doesn't start right unless you run `hx` from the root directory of you elm project, so `hx src/Test.elm`
3. Type the following into helix:

```elm
module Test exposing (..)

hi = case Expr of
```
4. Press `Enter` or `Enter + Tab`

I expected this to happen:
- Helix to keep working (with possible syntax highlighting issues)

Instead, this happened:
- `Segmentation fault (core dump)`

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
...
2023-05-06T19:03:00.835 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-05-06T19:03:01.027 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"range":{"end":{"character":17,"line":2},"start":{"character":17,"line":2}},"text":"\n"}],"textDocument":{"uri":"file:///home/ken/src/fun/elm/testing/ast-test/src/Test.elm","version":4}}}
2023-05-06T19:03:01.028 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-05-06T19:03:01.029 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"type":3,"message":"Changed text document, going to parse it. file:///home/ken/src/fun/elm/testing/ast-test/src/Test.elm"}}
2023-05-06T19:03:01.030 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, Notification(Notification { jsonrpc: Some(V2), method: "window/logMessage", params: Map({"message": String("Changed text document, going to parse it. file:///home/ken/src/fun/elm/testing/ast-test/src/Test.elm"), "type": Number(3)}) })))
2023-05-06T19:03:01.030 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Changed text document, going to parse it. file:///home/ken/src/fun/elm/testing/ast-test/src/Test.elm" }
2023-05-06T19:03:01.030 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
```

- I'm not sure that there was anything helpful for this issue in the log message.
- This is just the tail end of a very long output from `RUST_BACKTRACE=1 hx -vv src/Test.elm` 

</details>

### Platform

Linux

### Terminal Emulator

tmux 3.2a / kitty 0.28.1

### Helix Version

helix 23.03

**Comment by the-mikedavis** [comment]
The segfault most likely comes from the scanner in [tree-sitter-elm](https://github.com/elm-tooling/tree-sitter-elm) (scanners are custom-written C/C++ files)

**Comment by thistent** [comment]
I thought it was probably tree-sitter related. It would be preferable if there was a way to keep that from crashing helix, though.

**Comment by maca** [comment]
Cloning this repo and attempting to launch helix on it segfaults helix for me as soon as I open the file picker

https://github.com/dillonkearns/elm-markdown

**Comment by pascalkuthe** [comment]
Yeah segfaukts like this are basically always issues in the tree-sitter grammar (and sometimes very rarely tree-sitter itself). We can't di anything about this here. It's also not possible to sandbox native code so a sigfault will always take down the entire processes.

There seems to be an open issue on elm-grammar repo si you should follow the progress there/post any new reproduction cases there
